File: api/reset.js
import { parseSpec } from '../core/turing_machine.js';

export default function handler(req, res) {
  try {
    const { spec, input = '' } = req.body;

    if (!spec) {
      return res.status(400).json({
        success: false,
        error: 'Especificação vazia'
      });
    }

    const { tm, error } = parseSpec(spec);

    if (error) {
      return res.status(400).json({
        success: false,
        error
      });
    }

    tm.reset(input);

    return res.status(200).json({
      success: true,
      machine: tm.toDict()
    });
  } catch (e) {
    return res.status(500).json({
      success: false,
      error: `Erro no servidor: ${e.message}`
    });
  }
}

File: api/reset.py
from core.turing_machine import parse_spec
import json
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))


def handler(event, context):
    """Reseta a máquina com nova entrada"""
    try:
        if isinstance(event.get('body'), str):
            body = json.loads(event['body'])
        else:
            body = event.get('body', {})

        spec = body.get('spec', '')
        input_string = body.get('input', '')

        if not spec:
            return {
                'statusCode': 400,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps({
                    'success': False,
                    'error': 'Especificação vazia'
                })
            }

        tm, error = parse_spec(spec)

        if error:
            return {
                'statusCode': 400,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps({
                    'success': False,
                    'error': error
                })
            }

        tm.reset(input_string)

        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'success': True,
                'machine': tm.to_dict()
            })
        }

    except Exception as e:
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'success': False,
                'error': f'Erro no servidor: {str(e)}'
            })
        }
